alicat/serial.py

```diff
@@ -132,21 +132,175 @@ def set_gas(self, gas, retries=2):
         """Set the gas type.
 
         Args:
-            gas: The gas type, as a string. Supported gas types are:
+            gas: The gas type, as a string or integer. Supported gas types by string are:
                 'Air', 'Ar', 'CH4', 'CO', 'CO2', 'C2H6', 'H2', 'He', 'N2',
                 'N2O', 'Ne', 'O2', 'C3H8', 'n-C4H10', 'C2H2', 'C2H4',
                 'i-C2H10', 'Kr', 'Xe', 'SF6', 'C-25', 'C-10', 'C-8', 'C-2',
                 'C-75', 'A-75', 'A-25', 'A1025', 'Star29', 'P-5'
+
+                For the full gas table, please see page 52 of the controller manual here:
+                https://documents.alicat.com/manuals/DOC-MANUAL-MC.pdf
+
+                Gas mixes may only be called by their mix number.
         """
         self._test_controller_open()
 
-        if gas not in self.gases:
-            raise ValueError("{} not supported!".format(gas))
-        command = '{addr}$${gas}\r'.format(addr=self.address,
-                                           gas=self.gases.index(gas))
+        if isinstance(gas, str) is False:
+            num = True
+        else:
+            num = False
+
+        if num is True:
+            command = '{addr}$${index}\r'.format(addr=self.address,
+                                               index=gas)
+            self._write_and_read(command, retries)
+        else:
+            if gas not in self.gases:
+                raise ValueError("{} not supported!".format(gas))
+            command = '{addr}$${gas}\r'.format(addr=self.address, gas=self.gases.index(gas))
+            self._write_and_read(command, retries)
+
+        read_reg46 = self._write_and_read('{addr}$$R46\r'.format(addr=self.address), retries)
+        reg46 = int(read_reg46.split()[-1])
+        bits = [32768, 16384, 8192, 4096, 2048, 1024, 512]
+
+        for u in range(0, 7):
+            reg46 = reg46 - bits[u]
+            if reg46 < 0:
+                reg46 = reg46 + bits[u]
+            elif reg46 < 255:
+                break
```

Comment:
There is usually a cleaner one line way of bitmashing. This looks something like

```python
reg46 = int(read_reg46.split()[-1]) & 0b0000000111111111
# or even
reg46 = int(read_reg46.split()[-1]) & 0x1ff
```

The hex is shorter but the binary shows which bits are being turned off (if that's how you're thinking about this). Choose whichever one you find more readable and clear.